data/chat.js

```diff
@@ -0,0 +1,221 @@
+
+// Imports
+const { ObjectId } = require('mongodb');
+const mongoCollections = require('./../config/mongoCollections');
+const mongoConnection = require('./../config/mongoConnection');
+const chatsCol = mongoCollections.chats;
+const usersCol = mongoCollections.users;
+const relationshipsCol = mongoCollections.relationships;
+
+// Exported Functions
+module.exports = {
+    description: 'These are the functions for the chat feature of MentorMe',
+
+    /*
+    * Returns the inserted message object
+    * 
+    * Parameters:
+    *  - sender             string form of a user's _id
+    *  - relationship       string form of a relationship's _id
+    *  - message            string
+    */
+    newMessage: async (sender, relationship, message) => {
+        // <ERROR CHECKING>
+
+        // Check that sender is provided, it is a string, it is valid ObjectId, and that it exists in the database
+        if(!sender) throw "newMessage: sender must be provided";
+        if(typeof sender !== "string") throw "newMessage: sender must be a string";
+        if(!ObjectId.isValid(sender)) throw "newMessage: sender must be a valid ObjectId";
+        const usersCollection = await usersCol();
+        const senderUser = await usersCollection.findOne({ "_id": ObjectId(sender) });
+        if(senderUser === null) throw "newMessage: sender must be an existing user";
+
+        // Check that relationship is provided, it is a string, it is a valid ObjectId, it can be found in the database, and it includes the sender
+        if(!relationship) throw "newMessage: relationship must be provided";
+        if(typeof relationship !== "string") throw "newMessage: relationship must be a string";
+        if(!ObjectId.isValid(relationship)) throw "newMessage: relationship must be a valid ObjectId";
+        const relationshipsCollection = await relationshipsCol();
+        const messageRelationship = await relationshipsCollection.findOne({ "_id": ObjectId(relationship) });
+        if(messageRelationship === null) throw "newMessage: relationship must be an existing relationship";
+        let found = false;
+        if(messageRelationship["mentor"].toString().localeCompare(sender) === 0 || messageRelationship["mentee"].toString().localeCompare(sender) === 0) found = true;
+        if(!found) throw "newMessage: relationship must include sender";
+
+        // Check that message is provided, it is a string, it isn't empty, it isn't just spaces, and it isn't over the maximum message length
+        if(typeof message === "undefined") throw "newMessage: message must be provided";
+        if(typeof message !== "string") throw "newMessage: message must be a string";
+        if(message.length === 0) throw "newMessage: message must not be empty";
+        message = message.trim();
+        if(message.length === 0) throw "newMessage: message must not be just spaces";
+        if(message.length > MAX_MESSAGE_LENGTH) throw "newMessage: message must not be too long";
```

Comment:
That's an intensive error check.
Can you please move this part to some file under the validations folder?
 That helps in 
1.  distinguishing the business core code from validation code
2. Reusing the validations logic in your client-side js and Routes as well